tests: Bluetooth: Tester: BTP MTU handling improvements

sjanc opened the pull request:
This is not allowed by C11 and may cause issues if GNU extensions are disabled.

sjanc committed:
tests: Bluetooth: Tester: Avoid union with packed structure

This may cause some compiler warnings due to flexible array being
used in btp_hdr. Simply avoid it and do explicit pointer type cast
where needed.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

Thalley commented on tests/bluetooth/tester/src/btp.c:
```diff
@@ -31,11 +31,8 @@ static struct k_thread cmd_thread;
-		struct btp_hdr hdr;
-	};
+	uint8_t data[BTP_MTU]; /* includes btp header */
+	uint8_t rsp[BTP_MTU - sizeof(struct btp_hdr)];
```
Consider defining a `BTP_RSP_MTU` or something.

sjanc replied:
RSP has same MTU, it is just that we don't have (need) header in response buffer (since data from command header is used)

Thalley commented on tests/bluetooth/tester/src/btp.c:
```diff
@@ -90,25 +87,27 @@ static void cmd_handler(void *p1, void *p2, void *p3)
 		uint16_t len;
 
 		cmd = k_fifo_get(&cmds_queue, K_FOREVER);
+		hdr = (struct btp_hdr *)cmd->data;
```
Should we have an 
```suggestion
		hdr = (struct btp_hdr *)cmd->data;
		__ASSERT_NO_MSG(hdr->len <= BTP_DATA_MAX_SIZE);
```

sjanc replied:
actually, I'm not sure if this should be assert as this means we assert on invalid command,  I'll make this runtime check and simply return error (like we do for unknown command

Thalley commented:
The commit message doesn't really make sense anymore since `hdr` is not longer in the middle of the struct

sjanc committed:
tests: Bluetooth: Tester: Improve BTP MTU validation

Make sure MTU is validated to avoid reading pass command buffer.
Also make more explicit check in response length validation.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

Thalley commented on tests/bluetooth/tester/src/btp.c:
```diff
@@ -90,29 +87,33 @@ static void cmd_handler(void *p1, void *p2, void *p3)
-			if (btp->index != cmd->hdr.index) {
+			uint16_t len = sys_le16_to_cpu(hdr->len);
+
+			if (len > BTP_DATA_MAX_SIZE) {
+				status = BTP_STATUS_FAILED;
+			} else if (btp->index != hdr->index) {
```
Hmm, now I wonder if it's really need as we do have `} else if ((btp->expect_len >= 0) && (btp->expect_len != len)) {` which I guess would already catch it?

sjanc replied:
expect_len is used only for fixed size commands,  so it could stil pass it if command is variable length

Thalley replied:
Good point :) Could introduce something like a `btp->max_len` as well, but I guess this is more generic

sjanc replied:
yeah, we can do something more generic than BTP_HANDLER_LENGTH_VARIABLE, so that handlers for variable length commands would only have to validate variable part of command
